### PR TITLE
Adding Picard's RevertSam tool.

### PR DESCRIPTION
**Pull request by nh13**

### CODE REVIEW

File: tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import scala.collection.mutable.ListBuffer
+
+class RevertSam(val in: PathToBam,
+                val out: PathToBam,
+                sortOrder: SortOrder = SortOrder.queryname,
+                restoreOriginalQualities: Boolean = true,
+                removeDuplicateInformation: Boolean = true,
+                removeAlignmentInformation: Boolean = true,
+                attributesToClear: Option[Seq[String]] = None,
```
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala]: I get why you did this, but `Option[Seq[String]]` feels weird.  How would you feel about making it just `Seq[String]` with a default value of `new picard.sam.RevertSam().ATTRIBUTES_TO_CLEAR.toSeq`?

File: tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                santize: Boolean = false)
+  extends PicardTask with PicardMetricsTask {
```
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala]: I don't think this should include `PicardMetricsTask`.

File: tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  requires(Cores(1), Memory("4G"))
+
+  override def metricsExtension: String = MarkDuplicates.MetricsExtension
```
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala]: And this is probably a copy/paste error?

File: tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  override protected def addPicardArgs(buffer: ListBuffer[Any]): Unit = {
+    buffer.append("I=" + in)
+    buffer.append("O=" + out)
+    buffer.append("SO=" + sortOrder)
+    buffer.append("RESTORE_ORIGINAL_QUALITIES=" + restoreOriginalQualities)
+    buffer.append("REMOVE_DUPLICATE_INFORMATION=" + removeDuplicateInformation)
+    buffer.append("REMOVE_ALIGNMENT_INFORMATION=" + removeAlignmentInformation)
+    if (attributesToClear.nonEmpty) {
+      buffer.append("ATTRIBUTE_TO_CLEAR=" + null)
```
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala]: I'm a little leary of passing `null` to `+` here.  How about just `buffer.append("ATTRIBUTE_TO_CLEAR=null")`?

**Comment by tfenne** [comment]
A few minor comments @nh13.  Feel free to squash and merge when you've made the changes you feel are needed.

File: tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala (hunk 3 above)
Comment (nh13) [inline tasks/src/main/scala/dagr/tasks/picard/RevertSam.scala]: yup

**Pull request merged**